### PR TITLE
feat: initialize basic client setup 

### PR DESCRIPTION
**Pull request by abhimishra01**
### Description 

- generate client app boiler plate using [vite](https://vitejs.dev/)
- remove unwanted boiler plate related code
- use typescript based confiugrations

### CODE REVIEW

File: client/src/index.css [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  font-family: Inter, system-ui, Avenir, Helvetica, Arial, sans-serif;
+  line-height: 1.5;
+  font-weight: 400;
+
+  font-synthesis: none;
+  text-rendering: optimizeLegibility;
+  -webkit-font-smoothing: antialiased;
+  -moz-osx-font-smoothing: grayscale;
+  -webkit-text-size-adjust: 100%;
+}
```
Comment (BhanuPratap-Singh-Naruka) [inline client/src/index.css:11]: We should change these predefined styles.

**Review by BhanuPratap-Singh-Naruka — CHANGES_REQUESTED** [review]
We should delete vite.svg file from public folder

**Comment by abhimishra01** [comment]
> We should delete vite.svg file from public folder

for now lets keep it, until we get a good icon for our app as its used [here](https://github.com/JavascriptShinobis/messaging-app/blob/e2ff3f78824b3d8094c1e1cc297ca8f77e715450/client/index.html#L5)

File: client/src/index.css (hunk 1 above)
Comment (abhimishra01) [reply client/src/index.css]: > We should change these predefined styles.

done

**Review by BhanuPratap-Singh-Naruka — APPROVED** [review]

**Pull request merged**